src/id_list.ts

```diff
@@ -570,12 +570,55 @@ export class IdList {
    */
   delete(id: ElementId, count = 1) {
     checkCount(count);
+    if (count === 0) return this;
 
     // eslint-disable-next-line @typescript-eslint/no-this-alias
     let ans: IdList = this;
-    for (let i = 0; i < count; i++) {
-      ans = ans.deleteOne({ bunchId: id.bunchId, counter: id.counter + i });
+    let remaining = count;
+    let currentId = id;
```

Comment:
I would prefer if instead of these two variables, the loop just uses a `currentCounter` variable that holds currentId's counter. Compare that to `id.counter + count` in the loop condition.

Your code works as well, this is just a style preference.